nipype/interfaces/fsl/epi.py

```diff
@@ -788,30 +783,29 @@ def _gen_filename(self, name):
             return os.path.join(os.getcwd(), 'temp')
         return None
 
-    def _list_outputs(self):
-        outputs = self.output_spec().get()
+    def _post_run(self):
+        
         if not isdefined(self.inputs.exfdw):
-            outputs['exfdw'] = self._gen_filename('exfdw')
+            self.outputs.exfdw = self._gen_filename('exfdw')
         else:
-            outputs['exfdw'] = self.inputs.exfdw
+            self.outputs.exfdw = self.inputs.exfdw
         if isdefined(self.inputs.epi_file):
             if isdefined(self.inputs.epidw):
-                outputs['unwarped_file'] = self.inputs.epidw
+                self.outputs.unwarped_file = self.inputs.epidw
             else:
-                outputs['unwarped_file'] = self._gen_filename('epidw')
+                self.outputs.unwarped_file = self._gen_filename('epidw')
         if not isdefined(self.inputs.vsm):
-            outputs['vsm_file'] = self._gen_filename('vsm')
+            self.outputs.vsm_file = self._gen_filename('vsm')
         else:
-            outputs['vsm_file'] = self._gen_fname(self.inputs.vsm)
+            self.outputs.vsm_file = self._gen_fname(self.inputs.vsm)
         if not isdefined(self.inputs.tmpdir):
             outputs[
                 'exf_mask'] = self._gen_fname(cwd=self._gen_filename('tmpdir'),
                                               basename='maskexf')
         else:
-            outputs['exf_mask'] = self._gen_fname(cwd=self.inputs.tmpdir,
+            self.outputs.exf_mask = self._gen_fname(cwd=self.inputs.tmpdir,
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Unexpected keyword argument 'cwd' in method call](https://www.codacy.com/app/krzysztof-gorgolewski/nipy_nipype/file/2065943999/issues/source?bid=2767207&fileBranchId=2890884#l806)